packages/wrangler/src/create-worker-upload-form.ts

```diff
@@ -294,6 +295,14 @@ export function createWorkerUploadForm(worker: CfWorkerInit): FormData {
 		...(logpush !== undefined && { logpush }),
 	};
 
+	if (bindings.unsafe !== undefined && bindings.unsafe.metadata !== undefined) {
+		Object.keys(bindings.unsafe.metadata).forEach((key) => {
+			// eslint-disable-next-line @typescript-eslint/ban-ts-comment
+			// @ts-ignore
+			metadata[key] = bindings.unsafe.metadata[key];
+		});
```

Comment:
The problem you actually have here is that inside the `forEach()` callback TS cannot be sure that the `bindings.unsafe` property is still defined.

Assigning it to a constant variable solves that:

```suggestion
		const unsafeMetadata = bindings.unsafe.metadata;
		Object.keys(unsafeMetadata).forEach((key) => {
			metadata[key] = unsafeMetadata[key];
		});
```

Or alternatively avoid the callback altogether with a `for...of`:

```suggestion
		for(const key of Object.keys(bindings.unsafe.metadata)) {
			metadata[key] = bindings.unsafe.metadata[key];
		}
```